src/renderers/dom/shared/flattenStyleArray.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule flattenStyleArray
+ * @flow
+ */
+
+'use strict';
+
+type NestedStylesArray = Array<Object | NestedStylesArray | null>;
+
+function flattenStyleArray(nestedStylesArray: NestedStylesArray): Object {
+  var result = {};
+  var nestedStylesCopy = nestedStylesArray.slice();
+
+  var i = 0;
+  while (i < nestedStylesCopy.length) {
+    var styleObjOrArray = nestedStylesCopy[i];
+    if (Array.isArray(styleObjOrArray)) {
+      nestedStylesCopy.splice(i, 1, ...styleObjOrArray);
```

Comment:
This is a pretty expensive operation. A faster way is to recursively iterate on the array and update the `result` variable directly in the outer scope instead of mutating the array in place.